nomos-core/chain-defs/src/header/mod.rs

```diff
@@ -20,23 +22,36 @@ pub struct ContentId([u8; 32]);
 #[derive(Clone, Debug, Eq, PartialEq, Copy)]
 pub struct Nonce([u8; 32]);
 
+pub type Version = u8;
+
 #[derive(Clone, Debug, Serialize, Deserialize)]
 pub struct Header {
+    version: Version,
```

Comment:
let's use an enum with `repr(u8)` so that we get automatic verification during deserialization